modules/javafx.graphics/src/main/java/com/sun/glass/ui/Screen.java

```diff
@@ -389,16 +391,15 @@ public static void notifySettingsChanged() {
             for (Screen newScreen : screens) {
                 if (oldScreen.getNativeScreen() == newScreen.getNativeScreen()) {
                     w.setScreen(newScreen);
+                    oldScreens.add(oldScreen);
                     break;
                 }
             }
         }
 
-        // Dispose the old screens
-        if (oldScreens != null) {
-            for (Screen screen : oldScreens) {
-                screen.dispose();
-            }
+        // Dispose the old screens, if any
+        for (Screen screen : oldScreens) {
+            screen.dispose();
         }
```

Comment:
I'm still a little puzzled as to why a change was needed in the class at all. Even with the change you made to the native Windows glass code, it seems that calling dispose for all of the old Java screen objects is what we want. Am I missing something?